crates/nu-protocol/src/call_info.rs

```diff
@@ -45,12 +45,11 @@ impl EvaluatedArgs {
 
     /// Get the nth positional argument, error if not possible
     pub fn expect_nth(&self, pos: usize) -> Result<&Value, ShellError> {
+        let shell_error = ShellError::unimplemented("Better error: expect_nth");
```

Comment:
This means we always construct this error, even when we don't need it. Does something like this work:

```rust
self.positional
    .and_then(|array| array.get(pos).ok())
    .ok_or_else(|| ShellError::unimplemented("Better error: expect_nth"))
```

Reply:
Hmm, can't get this one to work.  Might be best for me to leave it as the original match.

Reply:
I reverted it for now, fixed the other `ok_or()`s to use `ok_or_else()`, and found a couple of other places where `ok_or()` would work.

Reply:
That's unfortunate, but I do know I sometimes have problems with closures and captures. Let me know if you want to ever come back to this and would like to pair on a solution!